TechRadarApi.Tests/V1/Gateways/TechnologyGatewayTests.cs

```diff
@@ -39,90 +68,70 @@ public async Task GetTechnologyByIdReturnsNullIfTechnologyDoesntExist()
             // Assert
+            var mockDynamoDb = new Mock<IDynamoDBContext>();
+            _classUnderTest = new TechnologyGateway(mockDynamoDb.Object);
             var id = Guid.NewGuid();
             var exception = new ApplicationException("Test Exception");
-            _dynamoDb.Setup(x => x.LoadAsync<TechnologyDbEntity>(id.ToString(), default))
+            mockDynamoDb.Setup(x => x.LoadAsync<TechnologyDbEntity>(id.ToString(), default))
                      .ThrowsAsync(exception);
             // Act
             Func<Task<Technology>> func = async () => await _classUnderTest.GetTechnologyById(id).ConfigureAwait(false);
             // Assert
             func.Should().Throw<ApplicationException>().WithMessage(exception.Message);
-            _dynamoDb.Verify(x => x.LoadAsync<TechnologyDbEntity>(id.ToString(), default), Times.Once);
+            mockDynamoDb.Verify(x => x.LoadAsync<TechnologyDbEntity>(id.ToString(), default), Times.Once);
         }
 
-        [Test]
-        [Ignore("Getting a bug - can't stub the DB response")]
+        [Fact]
         public async Task GetAllTechnologiesReturnsEmptyArrayIfNoTechnologiesExist()
         {
-            // Arrange
-            List<ScanCondition> conditions = new List<ScanCondition>();
-            var stubbedResponse = new List<TechnologyDbEntity>();
-
-            _dynamoDb.Setup(x => x.ScanAsync<TechnologyDbEntity>(conditions, default).GetRemainingAsync(default))
-                    .ReturnsAsync(stubbedResponse);
-            // Act
+            // Arrange + Act
             var response = await _classUnderTest.GetAll().ConfigureAwait(false);
             // Assert
-            _dynamoDb.Verify(x => x.ScanAsync<TechnologyDbEntity>(conditions, default).GetRemainingAsync(default), Times.Once);
             response.Should().BeEmpty();
         }
 
-        [Test]
-        [Ignore("Getting a bug - can't stub the DB response")]
+        [Fact]
         public async Task GetAllTechnologiesReturnsAnArrayOfAllTechnologiesInTheTable()
         {
             // Arrange
             var entities = _fixture.CreateMany<Technology>().ToList();
-            var stubbedResponse = entities.Select(x => DatabaseEntityHelper.CreateDatabaseEntityFrom(x)).ToList();
-
-            List<ScanCondition> conditions = new List<ScanCondition>();
-            _dynamoDb.Setup(x => x.ScanAsync<TechnologyDbEntity>(conditions, default).GetRemainingAsync(default))
-                    .ReturnsAsync(stubbedResponse);
+            SetupTestData(entities);
             // Act
             var response = await _classUnderTest.GetAll().ConfigureAwait(false);
             // Assert
-            _dynamoDb.Verify(x => x.ScanAsync<TechnologyDbEntity>(conditions, default).GetRemainingAsync(default), Times.Once);
             response.Should().BeEquivalentTo(entities);
         }
 
-        [Test]
-        [Ignore("Getting a bug - can't stub the DB response")]
-
+        [Fact(Skip = "Cannot mock DynamoDb context")]
```

Comment:
Why is this test skipped? what is the issue you are facing

Reply:
```
mockDynamoDb.Setup(x => x.ScanAsync<TechnologyDbEntity>(conditions, default).GetRemainingAsync(default))
                     .ThrowsAsync(exception);
```
This line ^ doesn't work with the scan async method as it's a 'non-virtual method'. Here's [a link](https://github.com/moq/moq4/issues/590) for more info. Should I just remove the test?